### PR TITLE
py*-beancount: upgrade to 2.3.4

### PR DESCRIPTION
**Pull request by sutra**
#### Description

Upgrade py*-beancount from 2.2.3 to 2.3.4.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.2 21D49 x86_64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-beancount/Portfile [hunk 1]
```diff
@@ -24,7 +24,8 @@ homepage            http://furius.ca/beancount/
 master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
 distname            ${python.rootname}-${version}
 
-python.versions     37 38
+python.versions     37 38 39 310
+python.default_version  38
```
Comment (pmetzger) [inline python/py-beancount/Portfile:28]: If possible, the default version should probably now be 3.10 (or the latest it will support.)

**Comment by mrdomino** [comment]
LGTM

**Comment by reneeotten** [comment]
> LGTM

no, it doesn't - the CI fails and also `port lint` gives warnings about missing dependencies for the Python 3.10 subport....

File: python/py-beancount/Portfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ homepage            http://furius.ca/beancount/
 master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
 distname            ${python.rootname}-${version}
```
Comment (reneeotten) [inline python/py-beancount/Portfile:25]: `master_sites` and `distname` can be removed, these are the defaults set by the `python` PG

**Comment by sutra** [comment]
> > LGTM
> 
> no, it doesn't - the CI fails and also `port lint` gives warnings about missing dependencies for the Python 3.10 subport....

Have to update the `py-bottle` and  `py-magic` to support python-310, or remove `310` from this `py-beancount`.

**Comment by sutra** [comment]
Squashed commits in #13898

**Comment by reneeotten** [comment]
please try to keep things in one PR as to have all comments retained. In the future, please just squash your commit locally and push to this PR.

**Comment by sutra** [comment]
> please try to keep things in one PR as to have all comments retained. In the future, please just squash your commit locally and push to this PR.

Thank you, I have learned and using in the new PRs about upgrading py-httplib2.

**Pull request closed without merging**